gremlin-javascript/src/main/javascript/gremlin-javascript/lib/utils.js

```diff
@@ -112,8 +95,36 @@ exports.getUserAgentHeader = function getUserAgentHeader() {
   return 'User-Agent';
 };
 
-const userAgent = generateUserAgent();
+exports.getUserAgent = async () => {
```

Comment:
Please excuse my lack of knowledge in JavaScript, but it seems like this changes the user agent header that we will be sending to the server. If this is the case, we will want to change this to be more like how the NodeUserAgent works. Our specification for the user agent can be found at https://tinkerpop.apache.org/docs/current/dev/provider/#_graph_driver_provider_requirements . Notably the form should be of "[Application Name] [GLV Name].[Version] [Language Runtime Version] [OS].[Version] [CPU Architecture]". The reason this is important is because the server will likely start using this value in the future to enable/disable features based on the GLV's version.

Reply:
Yes, but unfortunately to my knowledge, Web API Web Socket doesn't allow you to set custom headers.

Reply:
I see. Although it would have been best for this to work the same between the web version and the Node version, I still think its worth getting this PR in so I think this can be overlooked. By the way, in the master branch, we will be removing WebSocket support and transitioning to HTTP so this won't be an issue there anyway.